release.sh

```diff
@@ -98,95 +109,119 @@ if [[ "$CONFIRM" != "y" ]]; then
-
-# Update the version number in CDN URLs included in the README and the documentation,
-# and regenerate the Subresource Integrity hash for these files.
-node update-sri.js "${VERSION}" README.md contrib/*/README.md dist/README.md \
-    docs/*.md docs/*.md.bak website/pages/index.html
-
-# Generate a new version of the docs and publish the website
-pushd website
-yarn run version "${VERSION}"
-
-# Restore docs to use local built CSS
-for file in ../docs/*.md.bak; do
-    mv -f "$file" "${file%.bak}"
-done
 
-USE_SSH=true yarn publish-gh-pages
-popd
+if [[ $BRANCH != "master" ]]; then
+    # Build generated files
+    yarn build
+
+    # Regenerate the Subresource Integrity hash in the README and the documentation
+    node update-sri.js "${VERSION}" README.md contrib/*/README.md \
+        docs/*.md website/pages/index.html website/versioned_docs/version-$VERSION/*.md
+
+    # Make the commit and push
+    git add README.md contrib/*/README.md \
+        docs website/pages/index.html website/versioned_docs/
+    git commit -n -m "Update SRI hashes"
+    git push
+elif [[ ! $PUBLISH ]]; then
+    # Make a release branch
+    git checkout -b "v$VERSION-release"
+
+    # Edit package.json to the right version, as it's inlined (see
+    # http://stackoverflow.com/a/22084103 for why we need the .bak file to make
+    # this mac & linux compatible)
+    sed -i.bak -E 's|"version": "[^"]+",|"version": "'$VERSION'",|' package.json
+    rm -f package.json.bak
 
-# Make the commit and tag, and push them.
-git add package.json README.md contrib/*/README.md dist/README.md \
-    docs/*.md website/pages/index.html website/versioned_docs/ \
-    website/versioned_sidebars/ website/versions.json
-git commit -n -m "v$VERSION"
-git diff --stat --exit-code # check for uncommitted changes
-git tag -a "v$VERSION" -m "v$VERSION"
-git push origin "v$VERSION"
+    # Build generated files
+    yarn build
+
+    if [ ! -z "$NEXT_VERSION" ]; then
+        # Edit package.json to the next version
+        sed -i.bak -E 's|"version": "[^"]+",|"version": "'$NEXT_VERSION'-pre",|' package.json
+        rm -f package.json.bak
+        git add package.json
+    fi
+
+    # Edit docs to use CSS from CDN
+    grep -l '{@stylesheet: katex.min.css}' docs/*.md | xargs sed -i.bak \
+        's|{@stylesheet: katex.min.css}|<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@./dist/katex.min.css" integrity="sha384-katex.min.css" crossorigin="anonymous"/>|'
+
+    # Update the version number in CDN URLs included in the README and the documentation,
+    # and regenerate the Subresource Integrity hash for these files.
+    node update-sri.js "${VERSION}" README.md contrib/*/README.md \
+        docs/*.md docs/*.md.bak website/pages/index.html
+
+    # Generate a new version of the docs
+    pushd website
+    yarn run version "${VERSION}"
+    popd
+
+    # Restore docs to use local built CSS
+    for file in docs/*.md.bak; do
+        mv -f "$file" "${file%.bak}"
```

Comment:
`%.bak` means "without `.bak`"?

Reply:
@kevinbarabash Yes, it's bash substitution.